### DISCUSSION TITLE
DirtyFields is empty even if a input field is changed

### DISCUSSION THREAD

**Discussion by axisplay-neo** [react-hook-form/react-hook-form#2602] [Q&A]
When a field (input text) is changed, and the form is submitted, the dirtyFields is empty and isEmpty is false. Doesn't seems right to me. Please help. Thanks. 

[CodeSandBox.io](https://codesandbox.io/s/react-hook-form-reset-form-forked-4e8bm?file=/src/index.js:0-615)

**Comment by bluebill1049** [comment]
try to read form.dirtyFields before the `render`, we have optimized the render with Proxy.
![Screen Shot 2020-08-14 at 8 21 27 pm](https://user-images.githubusercontent.com/10513364/90239955-bce1c100-de6b-11ea-863f-f265edd15160.png)
